tests/integration/test_vulnerability_detector/test_feeds/canonical/test_extra_tags_canonical_feed.py

```diff
@@ -0,0 +1,96 @@
+
+# Marks
+pytestmark = pytest.mark.tier(level=2)
+
+current_test_path = os.path.dirname(os.path.realpath(__file__))
+test_data_path = os.path.join(current_test_path, '..', '..', 'data')
+configurations_path = os.path.join(test_data_path, 'configuration', 'test_feeds', vd.INVALID_CANONICAL_FEEDS_CONF)
+custom_canonical_oval_feed_path = os.path.join(test_data_path, 'feeds', vd.CUSTOM_CANONICAL_OVAL_FEED)
+
+wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+
+# Set configuration
+parameters = [{'CANONICAL_CUSTOM_FEED': custom_canonical_oval_feed_path}]
+ids = ['CANONICAL_configuration']
+
+test_data = [[1, 2, 3], {"a": 1, "b": 2}, "extra_tag", 12345, "ñ", "テスト", "ИСПЫТАНИЕ", "测试", "اختبار", " ", ""]
+test_values = [(tag, value) for tag in test_data for value in test_data]
+
+tags_ids = [f"tag: {value[0]}, value: {value[1]}" for value in test_values]
+
+# Insert extra tags before <generator> tag
+insert_pattern = r'(.*)(<generator>)(.*)'
+
+# Configuration data
+configurations = load_wazuh_configurations(configurations_path, __name__, params=parameters)
+
+
+@pytest.fixture(scope='module', params=configurations, ids=ids)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.fixture
+def modify_feed(test_values, request):
+    """
+    Modify the Canonical OVAL feed, setting a test tag value
+    """
+    backup_data = file.read_xml_file(file_path=custom_canonical_oval_feed_path, namespaces=vd.XML_FEED_NAMESPACES)
+
+    modified_data = insert_xml_tag(pattern=insert_pattern, tag=test_values[0], value=test_values[1],
+                                   data=str(backup_data))
+
+    file.write_file(file_path=custom_canonical_oval_feed_path, data=modified_data)
+
+    vd.clean_vuln_and_sys_programs_tables()
+
+    control_service('restart', daemon='wazuh-modulesd')
+
+    vd.set_system(system='BIONIC')
+
+    yield
+
+    file.write_file(file_path=custom_canonical_oval_feed_path, data=backup_data)
+
+    vd.clean_vuln_and_sys_programs_tables()
+
+    file.truncate_file(LOG_FILE_PATH)
+
+
+def test_no_feed_changes(clean_vuln_tables, get_configuration, configure_environment, restart_modulesd):
+    """Check if the feed is imported successfully by default"""
+    vd.check_feed_imported_successfully(wazuh_log_monitor=wazuh_log_monitor, log_system_name=vd.BIONIC_LOG,
+                                        expected_vulnerabilities_number=vd.CANONICAL_NUM_CUSTOM_VULNERABILITIES)
+
+
+@pytest.mark.parametrize('test_values', test_values, ids=tags_ids)
+def test_extra_tags_canonical_feed(test_values, clean_vuln_tables, get_configuration, configure_environment,
+                                   modify_feed):
+    """
+    Check if vulnerability detector behaves as expected when importing Canonical OVAL feed with extra tags
+    """
+    inserted_tag = test_values[0]
+
+    if inserted_tag != ' ' and type(inserted_tag) is str or type(inserted_tag) is int:
```

Comment:
```suggestion
    if inserted_tag != ' ' and type(inserted_tag) in [str, int]:
```

Reply:
Done in ba7bcc0729d9649d5d646f0d5ea80a440267f65b